plugins/default/lib/memory_usage.py

```diff
@@ -0,0 +1,123 @@
+#!/usr/bin/env python
+import argparse
+import sys
+import traceback
+from collections import Counter
+
+import openshift
+import nagios
+
+
+def generate_parser():
+    parser = argparse.ArgumentParser(
+        description="Checks container memory usage",
+    )
+    parser.add_argument(
+        "-w", "--warn", type=int, required=True,
+        help="set warning threshold percentage",
+    )
+    parser.add_argument(
+        "-c", "--crit", type=int, required=True,
+        help="set critical threshold percentage, "
+             "must be higher than or equal the warning threshold",
+    )
+    parser.add_argument(
+        "-p", "--project", required=False,
+        help="openshift project/namespace to use",
+        )
+    return parser
+
+
+check_memory_usage_cmd = ("cat", "/sys/fs/cgroup/memory/memory.usage_in_bytes")
+check_memory_limit_cmd = ("cat", "/sys/fs/cgroup/memory/memory.limit_in_bytes")
+
+
+def analize(pod, container, memory_limit, memory_used, warning_threshold, critical_threshold):
+    results = []
+    usage = (float(memory_used) / float(memory_limit)) * 100
+
+    nagios_status = nagios.UNKNOWN
+    if usage >= critical_threshold:
+        nagios_status = nagios.CRIT
+    elif usage >= warning_threshold:
+        nagios_status = nagios.WARN
+    elif usage < warning_threshold:
+        nagios_status = nagios.OK
+
+    results.append([pod, container, int(memory_limit), int(memory_used), usage, nagios_status])
+    return results
+
+
+def report(results, errors):
+    if not results:
+        return nagios.UNKNOWN
+
+    unique_statuses = Counter(
+        status
+        for pod, container, memory_total, memory_used, usage, status in results
+    )
+
+    ret = max(unique_statuses)
+
+    if ret == nagios.OK:
+        print "%s: All %s containers are under the warning threshold" % (
+            nagios.status_code_to_label(ret), len(results))
+    elif ret == nagios.UNKNOWN:
+        print "%s: Unable to determine usage on %s containers" % (
+            nagios.status_code_to_label(ret), unique_statuses[nagios.UNKNOWN])
+    elif ret == nagios.WARN:
+        print "%s: There are %s containers over the warning threshold" % (
+            nagios.status_code_to_label(ret), unique_statuses[nagios.WARN])
+    else:
+        print "%s: There are %s containers over the critical threshold and %s containers over the warning threshold" % (
+            nagios.status_code_to_label(ret), unique_statuses[nagios.CRIT], unique_statuses[nagios.WARN])
+
+    for pod, container, memory_total, memory_used, usage, status in results:
+        print "%s: %s: - total bytes: %s, used bytes: %s, usage: %.1f%%" % (
```

Comment:
Would it be nicer to use something like [this](http://stackoverflow.com/questions/1094841/reusable-library-to-get-human-readable-version-of-file-size) to output these bytes in human-readable format?